File: test-suite/handwritten-src/js/ConstantsTest.js
class ConstantsTest {
    constructor(module) {
        this.m = module;
    }
    testConstPrimitives() {
        assertEq(this.m.testsuite.Constants.I8_CONSTANT, 1);
        assertEq(this.m.testsuite.Constants.OPT_I8_CONSTANT, 1);
        assertEq(this.m.testsuite.Constants.STRING_CONSTANT, "string-constant");
    }
    testConstRecords() {
        assertEq(this.m.testsuite.Constants.OBJECT_CONSTANT, {
            someInteger: 3,
            someString: "string-constant"
        });
    }
}

allTests.push(ConstantsTest);

File: test-suite/handwritten-src/swift/ConstantsTest.swift
import XCTest
import DjinniSupport
@testable import TestSuite

final class ConstantsTest: XCTestCase {
    func testConstPrimitives() throws {
        XCTAssertEqual(Constants.i8Constant, 1)
        XCTAssertEqual(Constants.optI8Constant, 1)
        XCTAssertEqual(Constants.stringConstant, "string-constant")
    }

    func testConstRecords() throws {
        XCTAssertEqual(Constants.objectConstant, ConstantRecord(someInteger:3, someString:"string-constant"))
    }
}
